server/controllers/response.ts

```diff
@@ -155,3 +156,44 @@ export function getAllResponse(surveyId: string, done: (err: any, res?: Response
         }
     });
 }
+export async function downloadRaw (req: Request, res: Response, next: NextFunction) {
+    if (!req.user) { // if req.user is undefined
+        throw Error("Unreachable: this route handler is called only when the user is logged in");
+    }
+    const id = req.params.id;
+    const questionsCol = await Survey.findById(id);
```

Comment:
After this, we might want to check if the requesting user actually owns the survey. You can check some other route handlers from the survey controller file to see how that works. From the top of my head, it would look something like this.

```ts
if (!questionsCol || !req.user._id.equals(questionsCol.owner)) {
    // Maybe redirect?
    return res.redirect(...);
}
```

Since we also check if the queried survey actually exists, you'll be able to remove `?` signs after `questionsCol` below.